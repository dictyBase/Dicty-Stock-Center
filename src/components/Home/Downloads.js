// @flow
import React from "react"
import LinkList from "../LinkList"
import { withStyles } from "@material-ui/core/styles"
import styles from "./homeStyles"

const downloadLinks = [
  { name: "Phenotype Ontology", to: "", routerAware: true },
  { name: "Strain Characteristics", to: "", routerAware: true },
  { name: "Mutagenesis Methods", to: "", routerAware: true },
  { name: "Plasmid Keywords", to: "", routerAware: true },
]

type Props = {
  /** Material-UI styling */
  classes: {
    panelGray: string,
  },
}

/**
 * Downloads displays download links on the homepage.
 */

const Downloads = (props: Props) => {
  const { classes } = props

  return (
    <div className={classes.panelGray}>
      <h3>Download / View</h3>
      <LinkList list={downloadLinks} />
    </div>
  )
}

export default withStyles(styles)(Downloads)
