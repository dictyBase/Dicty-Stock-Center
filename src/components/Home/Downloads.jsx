import React, { Component } from "react"
import LinkList from "../LinkList"
import { PanelBlue } from "styles"

export default class Downloads extends Component {
  displayName = "downloads links"
  render() {
    const downloads = [
      { name: "Phenotype Ontology", to: "", routerAware: true },
      { name: "Strain Characteristics", to: "", routerAware: true },
      { name: "Mutagenesis Methods", to: "", routerAware: true },
      { name: "Plasmid Keywords", to: "", routerAware: true }
    ]
    return (
      <PanelBlue>
        <LinkList list={downloads} title="Download / View" />
      </PanelBlue>
    )
  }
}
