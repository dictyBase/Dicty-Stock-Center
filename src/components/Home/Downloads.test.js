import React from "react"
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import Downloads from "./Downloads"

test("matching a snapshot of Downloads", () => {
  const component = renderer.create(
    <BrowserRouter>
      <Downloads />
    </BrowserRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
