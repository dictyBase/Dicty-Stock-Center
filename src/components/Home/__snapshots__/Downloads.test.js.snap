// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Downloads 1`] = `
.c0 {
  background-color: #eff8fb;
  border: 1px solid #eff8fb;
  margin: 15px auto;
  padding: 15px;
  border-radius: 10px;
  width: 100%;
}

.c1 ul,
.c1 li {
  margin: 0;
  padding: 0;
  list-style: none;
}

.c1 a {
  font-size: 18px;
}

<div
  className="c0"
>
  <h3>
    Download / View
  </h3>
  <div
    className="c1"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Phenotype Ontology
        </a>
      </li>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Strain Characteristics
        </a>
      </li>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Mutagenesis Methods
        </a>
      </li>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Plasmid Keywords
        </a>
      </li>
    </ul>
  </div>
</div>
`;
